Lib/pdb.py

```diff
@@ -1315,7 +1315,14 @@ def do_commands(self, arg):
         reached.
         """
         if not arg:
-            bnum = len(bdb.Breakpoint.bpbynumber) - 1
+            for bp in reversed(bdb.Breakpoint.bpbynumber):
+                if not bp:
```

Comment:
Are you checking this against `None`? We prefer to do it explicitly with `is not None`. There are too many falsy values and we want to be as clear as possible.